R/Model-class.R

```diff
@@ -185,8 +197,12 @@ ModelLogNormal <- function(mean, cov, ref_dose = 1) {
       alpha0 <- theta[1]
       alpha1 <- exp(theta[2])
     },
-    modelspecs = function() {
-      list(ref_dose = ref_dose, mean = params@mean, prec = params@prec)
+    modelspecs = function(from_prior) {
+      if (from_prior) {
+        list(mean = params@mean, prec = params@prec)
+      } else {
+        list(ref_dose = ref_dose, mean = params@mean, prec = params@prec)
+      }
```

Comment:
```suggestion
      ret <- list(mean = params@mean, prec = params@prec)
      if (!from_prior) {
        ret$ref_dose <- ref_dose
      } 
      ret
```

Reply:
this makes it explicit that we have additional stuff being added to the return list when it is not from prior.

Reply:
Sure, I was even thinking about it but finally chose something different :)